src/cache.ts

```diff
@@ -79,15 +80,13 @@ export class ListWatch<T extends KubernetesObject> implements ObjectCache<T>, In
         return this.resourceVersion;
     }
 
-    private async errorHandler(err: any): Promise<void> {
+    private async doneHandler(err: any): Promise<any> {
+        if (this.request) {
```

Comment:
make this a call to `this.stop()` so that we share that code.

Reply:
good point. will do 👍

Reply:
fixed